### PR TITLE
PA-2238 Replace ParentGroupedFilter with ParentSelect

### PR DESCRIPTION
**Pull request by agalbrai**

### CODE REVIEW

File: frontend/src/components/common/form/ParentSelect.tsx [hunk 1]
```diff
@@ -57,13 +57,15 @@ export const ParentSelect: React.FC<IParentSelect> = ({
...
     if (!loaded) {
-      if (value?.value) {
-        setFieldValue(field, value);
-      } else if (value && !value.value) {
-        setFieldValue(field, option);
-      } else {
-        setFieldValue(field, option);
-        setLoaded(true);
+      if (value !== undefined) {
+        if (value?.value) {
```
Comment (agalbrai) [inline frontend/src/components/common/form/ParentSelect.tsx:61]: A test kept failing unless this was added, I can't remember what the reasoning was but @jdtoombs figured it out

File: frontend/src/features/admin/users/ManageUsers.test.tsx [hunk 2]
```diff
@@ -103,12 +106,7 @@ describe('Manage Users Component', () => {
         },
       });
     });
-    expect(getAllByRole('option')[0]).toHaveTextContent('agencyVal');
-  });
-
-  it('Does not display disabled agencies', () => {
-    const { queryByText } = testRender(getStore());
-    expect(queryByText('disabledAgency')).toBeNull();
```
Comment (agalbrai) [inline frontend/src/features/admin/users/ManageUsers.test.tsx:111]: For whatever reason the test can't actually see the dropdown options with ParentSelect, so this test wasn't actually testing anything

File: frontend/src/components/common/form/ParentSelect.tsx (hunk 1 above)
Comment (devinleighsmith) [reply frontend/src/components/common/form/ParentSelect.tsx:61]: I don't really see why the if statement on 60 is required, unless the else statement shouldn't execute when value is undefined?

File: frontend/src/features/admin/users/ManageUsers.test.tsx (hunk 2 above)
Comment (devinleighsmith) [reply frontend/src/features/admin/users/ManageUsers.test.tsx:111]: something to try in situations like this is dumping the entire dom with prettydom (something like console.log(prettyDOM(container, 99999))

alternatively sometimes you need to wait for your condition to be true:
```
wait(() => {
//wait will run and expectations in a loop until it times out or the expectation stops failing.
  expect(true).toBeTruthy();
});
```

**Review by devinleighsmith — APPROVED** [review]

File: frontend/src/components/common/form/ParentSelect.tsx (hunk 1 above)
Comment (jdtoombs) [reply frontend/src/components/common/form/ParentSelect.tsx:61]: I think it does throw an error threw the test if I remember correctly, I feel like this logic could be rearranged a bit better (from what I did originally). But I have a ticket for cleaning this component up/fixing warnings so I could do it then!

**Review by jdtoombs — APPROVED** [review]

File: frontend/src/components/common/form/ParentSelect.tsx (hunk 1 above)
Comment (Fosol) [reply frontend/src/components/common/form/ParentSelect.tsx:61]: `if (value?.value)` will return false if `value === undefined`, so I'm not sure why the need for `if (value !== undefined)`

**Review by Fosol — APPROVED** [review]

**Pull request merged**